lib/src/shared/main/java/com/couchbase/lite/internal/fleece/FLArray.java

```diff
@@ -20,72 +20,69 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import com.couchbase.lite.internal.utils.Preconditions;
+import com.couchbase.lite.utils.Fn;
+
 
 public class FLArray {
+    private final long handle; // pointer to FLArray
+
+    //-------------------------------------------------------------------------
+    // constructor
+    //-------------------------------------------------------------------------
+
+    public FLArray(long handle) {
+        Preconditions.checkArgNotZero(handle, "handle");
+        this.handle = handle;
+    }
+
+    //-------------------------------------------------------------------------
+    // public methods
+    //-------------------------------------------------------------------------
+
     /**
-     * Returns the number of items in an array, or 0 if the pointer is nullptr.
+     * Returns the number of items in an array; 0 if handle is null.
      *
-     * @param array FLArray
-     * @return long (uint32_t)
+     * @return the number of items in an array; 0 if handle is null.
      */
-    static native long count(long array);
+    public long count() { return count(handle); }
 
     /**
-     * Returns an value at an array index, or nullptr if the index is out of range.
+     * Returns an value at an array index, or null if the index is out of range.
      *
-     * @param array FLArray
-     * @param index uint32_t
-     * @return long (FLValue)
+     * @param index index for value
+     * @return the FLValue at index
      */
-    static native long get(long array, long index);
-
-    private final long handle; // hold pointer to FLArray
-
-    //-------------------------------------------------------------------------
-    // public methods
-    //-------------------------------------------------------------------------
-    public FLArray(long handle) {
-        this.handle = handle;
-    }
+    public FLValue get(long index) { return new FLValue(get(handle, index)); }
 
     public List<Object> asArray() { return asTypedArray(); }
 
     @SuppressWarnings("unchecked")
     public <T> List<T> asTypedArray() {
         final List<T> results = new ArrayList<>();
         final FLArrayIterator itr = new FLArrayIterator();
-        try {
-            itr.begin(this);
-            FLValue value;
-            while ((value = itr.getValue()) != null) {
-                results.add((T) value.asObject());
-                if (!itr.next()) { break; }
-            }
-        }
-        finally {
-            itr.free();
+
+        itr.begin(this);
+        FLValue value;
+        while ((value = itr.getValue()) != null) {
+            results.add((T) value.asObject());
+            if (!itr.next()) { break; }
         }
-        return results;
-    }
 
-    public long count() {
-        return count(handle);
+        return results;
```

Comment:
Don't you need to free FLArrayIterator after usage?

Reply:
Same deal.  Its finalizer should take care of that...